mempool/cat/reactor.go

```diff
@@ -357,13 +355,31 @@ func (memR *Reactor) Receive(e p2p.Envelope) {
 			return
 		}
 
-		expectedSeq, haveExpected := memR.sequenceExpectationForSigner(msg.Signer, false, false)
+		expectedSeq, haveExpected := memR.querySequenceFromApplication(msg.Signer)
```

Comment:
```suggestion
		expectedSeq, haveSequence := memR.querySequenceFromApplication(msg.Signer)
```

haveSequence feels more readable